libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -579,32 +587,48 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			else
 				args.emplace_back(convert(*arguments[i], *parameterTypes[i]).commaSeparatedList());
 
-		if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+		optional<FunctionDefinition const*> functionDef;
+		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		{
+			solUnimplementedAssert(!functionType->bound(), "Internal calls to bound functions are not yet implemented for libraries and not allowed for contracts");
+
+			functionDef = dynamic_cast<FunctionDefinition const*>(memberAccess->annotation().referencedDeclaration);
+			solUnimplementedAssert(functionDef.value() != nullptr, "Calls via pointers stored in state variables are not implemented");
```

Comment:
Actually to be honest - the whole construct here just tries to avoid the dispatch routine if we can statically determine the called function. Because of that, I think you can remove the assertion here. We can always fall back to the function identifier taken from a variable and that has to be implemented elsewhere. Actually I think calls to pointers stored in state variables might even be already fully implemented.

Reply:
One assertion you could add is `functionDef == functionType->declaration()`.

Reply:
They aren't (unimplemented error in `YulUtilFunctions`) but ok. I thought that the static call was the default, more important case but if we want "dispatch unless we can avoid it" then yeah, it makes sense. I'll remove it.

Reply:
`solUnimplementedAssert()` removed. New assertion added.